src/assets/TokensController.ts

```diff
@@ -213,102 +260,238 @@ export class TokensController extends BaseController<
       );
+      const { newAllTokens, newAllIgnoredTokens, newAllDetectedTokens } =
+        this._getNewAllTokensState({
+          newTokens,
+          newIgnoredTokens,
+          newDetectedTokens,
+        });
 
       this.update({
+        tokens: newTokens,
+        ignoredTokens: newIgnoredTokens,
+        detectedTokens: newDetectedTokens,
         allTokens: newAllTokens,
-        tokens,
         allIgnoredTokens: newAllIgnoredTokens,
+        allDetectedTokens: newAllDetectedTokens,
+      });
+      return newTokens;
+    } finally {
+      releaseLock();
+    }
+  }
+
+  /**
+   * Add a batch of tokens.
+   *
+   * @param tokensToImport - Array of tokens to import.
+   */
+  async addTokens(tokensToImport: Token[]) {
+    const releaseLock = await this.mutex.acquire();
+    const { tokens, detectedTokens, ignoredTokens } = this.state;
+    const importedTokensMap: { [key: string]: true } = {};
+    // Used later to dedupe imported tokens
+    const newTokensMap = tokens.reduce((output, current) => {
+      output[current.address] = current;
+      return output;
+    }, {} as { [address: string]: Token });
+
+    try {
+      tokensToImport.forEach((tokenToAdd) => {
+        const { address, symbol, decimals, image, aggregators } = tokenToAdd;
+        const checksumAddress = toChecksumHexAddress(address);
+        const formattedToken: Token = {
+          address: checksumAddress,
+          symbol,
+          decimals,
+          image,
+          aggregators,
+        };
+        newTokensMap[address] = formattedToken;
+        importedTokensMap[address.toLowerCase()] = true;
+        return formattedToken;
+      });
+      const newTokens = Object.values(newTokensMap);
+
+      const newDetectedTokens = detectedTokens.filter(
+        (token) => !importedTokensMap[token.address.toLowerCase()],
+      );
+      const newIgnoredTokens = ignoredTokens.filter(
+        (tokenAddress) => !newTokensMap[tokenAddress.toLowerCase()],
+      );
+
+      const { newAllTokens, newAllDetectedTokens, newAllIgnoredTokens } =
+        this._getNewAllTokensState({
+          newTokens,
+          newDetectedTokens,
+          newIgnoredTokens,
+        });
+
+      this.update({
+        tokens: newTokens,
+        allTokens: newAllTokens,
+        detectedTokens: newDetectedTokens,
+        allDetectedTokens: newAllDetectedTokens,
         ignoredTokens: newIgnoredTokens,
+        allIgnoredTokens: newAllIgnoredTokens,
       });
-      return tokens;
     } finally {
       releaseLock();
     }
   }
 
   /**
-   * Adds a batch of tokens to the stored token list.
+   * Ignore a batch of tokens.
    *
-   * @param tokensToAdd - Array of Tokens to be added or updated.
-   * @returns Current token list.
+   * @param tokensToIgnore - Array of tokens to ignore.
    */
-  async addTokens(tokensToAdd: Token[]): Promise<Token[]> {
+  async ignoreTokens(tokensToIgnore: Token[]) {
```

Comment:
This new function doesn't appear to be used anywhere, and it overlaps in functionality with `removeAndIgnoreToken`. What was this for?

Reply:
Deprecated this in favor of the new method `ignoreTokens`, which can handle any number of tokens.